File: app/src/main/java/com/example/dropex/fragments/CartFragment.java
package com.example.dropex.fragments;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.GridView;

import com.example.dropex.R;
import com.example.dropex.adpaters.CartGVAdapter;
import com.example.dropex.adpaters.HomeGVAdapter;
import com.example.dropex.enums.Size;
import com.example.dropex.models.CartModel;
import com.example.dropex.models.ProductModel;

import java.util.ArrayList;

public class CartFragment extends Fragment {

    GridView cartGVAdapter;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_cart, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        cartGVAdapter = view.findViewById(R.id.cartGV);
        ArrayList<CartModel> cartModelArrayList = new ArrayList<>();

        cartModelArrayList.add(new CartModel(1, 2, Size.l, 1, 2));
        cartModelArrayList.add(new CartModel(1, 1, Size.m, 1, 3));
        cartModelArrayList.add(new CartModel(1, 1, Size.xxl, 1, 4));

        CartGVAdapter adapter = new CartGVAdapter(view.getContext(), cartModelArrayList);
        cartGVAdapter.setAdapter(adapter);
    }
}

File: app/src/main/java/com/example/dropex/fragments/CartFragment.kt
package com.example.dropex.fragments

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.GridView
import androidx.fragment.app.Fragment
import androidx.lifecycle.lifecycleScope
import com.example.dropex.R
import com.example.dropex.adpaters.CartGVAdapter
import com.example.dropex.database.CartRepository
import com.example.dropex.database.ProductsRepository
import com.example.dropex.models.ProductModel
import com.example.dropex.screens.PaymentGatewayScreen
import com.google.android.material.button.MaterialButton
import kotlinx.coroutines.launch

class CartFragment : Fragment() {

    private lateinit var cartGV: GridView
    private var cartRepository: CartRepository = CartRepository()
    private var productsRepository: ProductsRepository = ProductsRepository()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_cart, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        cartGV = view.findViewById(R.id.cartGV)
        val checkoutBtn = view.findViewById<MaterialButton>(R.id.checkoutBtn)

        checkoutBtn.setOnClickListener {
            val intent = Intent(context, PaymentGatewayScreen::class.java)
            startActivity(intent)
        }

        viewLifecycleOwner.lifecycleScope.launch {
            val cartModelArrayList = cartRepository.getCartItems()
            val productModels = ArrayList<ProductModel?>()
            for (cartModel in cartModelArrayList) {
                productModels.add(productsRepository.getProductById(cartModel!!.productId))
            }
            val adapter = CartGVAdapter(view.context, cartModelArrayList, productModels)
            cartGV.adapter = adapter
        }
    }
}